grafana/influxdb_service_discovery.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
@vishh you asked a question about why this was needed. Grafana supports both proxy and direct methods to access InfluxDB backend. This is mainly for deployments where InfluxDB is exposed as an external service and can be accessed directly from the browser. This code looks up the external service URL for InfluxDB and returns it if one is available.